components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/SessionDataPublisherImpl.java

```diff
@@ -123,6 +123,19 @@
                         .getUserName(), e);
             }
         }
+        SystemApplicationDAO systemApplicationDAO = new SystemApplicationDAO();
+        try {
+            systemApplicationDTOS = systemApplicationDAO.getApplications(tenantDomain);
+            if (systemApplicationDTOS.length < 0) {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "The tenant: " + tenantDomain + "doesn't have any system apps");
+                }
+            }
+        } catch (APIMgtDAOException e) {
+            log.error("Error thrown while retrieving system applications for the tenant domain" + tenantDomain, e);
```

Comment:
Can we continue when error happens?

Reply:
Yes, then there won't be any tokens related to system apps will be revoked and will log that above error message, but will terminate the user session from IDP side and user will be logged out of the client. 
Therefore, if there is an issue in database connections, it will not affect the logout flow. The downside is, then the tokens related to that user also won't get revoked.